### ISSUE TITLE
How to only lint changed files

### ISSUE THREAD

**Issue by squarefrog** [steven851007/SwiftLint_build_phase_example#1]
Firstly, thanks for this example - SwiftLint takes 30+ seconds to run on our project, so it was a nightmare running each time.

We run a full SwiftLint / SwiftFormat check on every PR on CI, so for us the main benefit with a script like this is only running for new/modified files. This is fairly easy to change this script to support this.

First, you can edit the [`diff-filter`](https://github.com/steven851007/SwiftLint_build_phase_example/blob/main/generate_swifttlint_filelist.sh#L27) from `ADR` to `AM`. This will then only add files which have been added or removed to the input file list.

Secondly, you can edit the build phase script to only run against the added and removed files:

```bash
./SwiftLint/swiftlint $(cat $SCRIPT_INPUT_FILE_LIST_0)
```
Conveniently this replaces the newline characters in the file with spaces, so SwiftLint will only run on these files.

The last step needed is to modify the script so instead of returning ALL swift files, only the changed swift files from `git_diff_result` is used. I'll update this comment with changes later on.

I just thought this would be useful to document in case anyone else has this use-case. Feel free to close this issue as no action is needed 👍

**Comment by jgale-nuvoair** [comment]
Do you have that final modification @squarefrog ? I was originally under the impression that was the whole purpose of this script, but your comment helped me realize it was not.

**Comment by winkelsdorf** [comment]
Good point, I was too under the impression that this actually speeds up things. As SwiftLint does not support `xcfilelist`, the whole point of generating a changed file list is somehow meaningless, isn't it? Edit: Speaking only of this way generating it.

Crosslink: https://github.com/realm/SwiftLint/issues/5195

**Comment by squarefrog** [comment]
You can pass the result of the file list to swiftlint. This works fine but it ignores any excludes you have in your swiftlint config file, if I remember correctly.

**Comment by winkelsdorf** [comment]
@squarefrog Absolutely right, I am going to cat and pipe the content of the file list to swiftlint. And `--force-exclude` helps preserving the exclusions (despite exclusions merge being slow as mentioned here https://github.com/realm/SwiftLint/issues/5018).

**Comment by mijo-gracanin** [comment]
@squarefrog If `M` is used in `diff-filter`, you also need to compare hashes of modified files, right?
Does anyone have an improved fork?

**Comment by NikKovIos** [comment]
For now this repo works like "if no changed for any file in target - mark target not to lint. If any change, lint all target". Would be great to lint only modified files. I think it could be reached with `--use-script-input-files`.
